lib/core/src/display/navigation/camera_manager.rs

```diff
@@ -0,0 +1,197 @@
+    }
+
+    // Initialize mouse events.
+    fn initialize_events(&mut self) -> Result<()> {
+        self.disable_context_menu()?;
+        self.initialize_wheel_zooming()?;
+        self.initialize_interaction_start_event()?;
+        self.initialize_mouse_move_event()?;
+        self.initialize_interaction_end_event()
+    }
+
+    fn initialize_wheel_zooming(&mut self) -> Result<()> {
+        let data = self.data.clone();
+        self.mouse_manager.add_mouse_wheel_callback(move |event:MouseWheelEvent| {
+            let zoom_event = ZoomingEvent::from_mouse_wheel(event, &data);
+            let event      = Event::Zooming(zoom_event);
+            data.borrow_mut().event_queue.push(event);
+        })?.forget();
+        Ok(())
+    }
+
+    fn initialize_interaction_start_event(&mut self) -> Result<()> {
+        let data = self.data.clone();
+        self.mouse_manager.add_mouse_down_callback(move |event:MouseClickEvent| {
+            match event.button {
+                MouseButton::MIDDLE => {
+                    data.borrow_mut().movement_type = Some(MovementType::Panning)
+                },
+                MouseButton::RIGHT => {
+                    let focus = event.position;
+                    data.borrow_mut().movement_type  = Some(MovementType::Zooming { focus })
+                },
+                _ => ()
+            }
+        })?.forget();
+        Ok(())
+    }
+
+    fn disable_context_menu(&mut self) -> Result<()> {
+        self.mouse_manager.set_context_menu(State::Disabled)
+    }
+
+    fn initialize_interaction_end_event(&mut self) -> Result<()> {
+        let data    = self.data.clone();
+        let closure = move |_| data.borrow_mut().movement_type = None;
+        self.mouse_manager.add_mouse_up_callback(closure)?.forget();
+
+        let data = self.data.clone();
+        let closure = move |_| data.borrow_mut().movement_type = None;
+        self.mouse_manager.add_mouse_leave_callback(closure)?.forget();
+        Ok(())
+    }
+
+    fn initialize_mouse_move_event(&mut self) -> Result<()> {
+        let data = self.data.clone();
+        self.mouse_manager.add_mouse_move_callback(move |event:MousePositionEvent| {
+            let mut data = data.borrow_mut();
+            data.mouse_position = event.position;
+
+            if let Some(movement_type) = &data.movement_type {
+                match movement_type {
+                    MovementType::Zooming { focus } => {
+                        let zoom_event = ZoomingEvent::from_mouse_move(event, *focus);
+                        let event      = Event::Zooming(zoom_event);
+                        data.event_queue.push(event);
+                    },
+                    MovementType::Panning => {
+                        let pan_event      = PanningEvent::from(event);
+                        let event          = Event::Panning(pan_event);
+                        data.event_queue.push(event);
+                    }
```

Comment:
This is not very configurable.

Reply:
This can be extended when we have a clear requirement demanding it.